vellum/workflows/triage_message/nodes/fetch_contact_summary_node.py

```diff
@@ -0,0 +1,43 @@
+import os
+from typing import Optional
+from vellum.workflows.nodes import BaseNode
+from services.aws import get_aws_session
+from .read_message_node import ReadMessageNode
+
+
+class FetchContactSummaryNode(BaseNode):
+    message = ReadMessageNode.Outputs.message
+
+    class Outputs(BaseNode.Outputs):
+        current_summary: Optional[str]
+        contact_id: str
+
+    def run(self) -> Outputs:
+        contact_id = str(self.message.contact_id)
+        
+        try:
+            session = get_aws_session()
+            s3_client = session.client("s3")
+            bucket_name = "vargas-jr-memory"
+            
+            if os.getenv("VERCEL_ENV") == "production":
+                key = f"contacts/{contact_id}/summary.txt"
+            else:
+                preview_id = os.getenv("VERCEL_GIT_COMMIT_SHA", "preview")
+                key = f"previews/{preview_id}/contacts/{contact_id}/summary.txt"
```

Comment:
This if else tree that considers preview files should happen as its own services util that we import to format the key. I believe our web app does something similar